docs/NooBaaNonContainerized/ConfigFileCustomizations.md

```diff
@@ -0,0 +1,461 @@
+    4. all - for all detailed debugging.  
+* <u>Steps</u>:  
+    ```
+    1. Open /path/to/config_dir/config.json file.
+    2. Set the NOOBAA_LOG_LEVEL key to the desired level.
+    Example:
+    "NOOBAA_LOG_LEVEL": "nsfs"
+    3. systemctl restart noobaa
+    ```
+
+### 3. Ports -
+* <u>Keys</u>: `ENDPOINT_PORT`, `ENDPOINT_SSL_PORT`, `ENDPOINT_SSL_STS_PORT`, `EP_METRICS_SERVER_PORT`  
+* <u>Type</u>: Number  
+* <u>Default</u>: ENDPOINT_PORT=6001, ENDPOINT_SSL_PORT=6443, ENDPOINT_SSL_STS_PORT=7443, EP_METRICS_SERVER_PORT=7004  
+* <u>Description</u>: Adjust the ports used by the application for communication
+    The following S3 endpoint ports can be customized -
+    1. ENDPOINT_PORT - S3 HTTP port 
+    2. ENDPOINT_SSL_PORT - S3 HTTPS port 
+    3. ENDPOINT_SSL_STS_PORT - STS HTTPS port 
+    4. EP_METRICS_SERVER_PORT - Prometheus metrics port  
+* <u>Steps</u>:  
+    ```
+    1. Open /path/to/config_dir/config.json file.
+    2. Set the config key -
+    Example:
+    "ENDPOINT_PORT": 5555
+    3. systemctl restart noobaa
+    ```
+
+### 4. Allow HTTP requests-
+* <u>Key</u>: `ALLOW_HTTP`  
+* <u>Type</u>: Boolean  
+* <u>Default</u>: false  
+* <u>Description</u>: Enable or disable listening to HTTP requests.  
+* <u>Steps</u>:  
+    ```
+    1. Open /path/to/config_dir/config.json file.
+    2. Set the config key -
+    Example:
+    "ALLOW_HTTP": true
+    3. systemctl restart noobaa
+    ```
+
+### 5. MD5 calculation -
+* <u>Key</u>: `NSFS_CALCULATE_MD5`  
+* <u>Type</u>: Boolean   
+* <u>Default</u>: false  
+* <u>Description</u>: Enable or disable MD5 calculation on PUT object or Multipart upload requests.  
+* <u>Steps</u>:  
+    ```
+    1. Open /path/to/config_dir/config.json file.
+    2. Set the config key -
+    Example:
+    "NSFS_CALCULATE_MD5": true
+    ```
+
+### 6. UV_THREADPOOL_SIZE -
+* <u>Key</u>: `UV_THREADPOOL_SIZE`  
+* <u>Type</u>: Number  
+* <u>Default</u>: 4  
+* <u>Description</u>: Set UV_THREADPOOL_SIZE in order to increase the performance of NooBaa - https://docs.libuv.org/en/v1.x/threadpool.html.  
+* <u>Steps</u>:  
+    ```
+    1. Open /path/to/config_dir/config.json file.
+    2. Set the config key -
+    Example:
+    "UV_THREADPOOL_SIZE": 8
+    3. systemctl restart noobaa
+    ```
+
+### 7. GPFS library path -
+* <u>Key</u>: `GPFS_DL_PATH`  
+* <u>Type</u>: String  
+* <u>Default</u>: ''   
+* <u>Description</u>: Set GPFS library path location.  
+* <u>Steps</u>:  
+    ```
+    1. Open /path/to/config_dir/config.json file.
+    2. Set the config key -
+    Example:
+    "GPFS_DL_PATH": "/usr/lib64/libgpfs.so"
+    3. systemctl restart noobaa
+    ```
+
+### 8. Buffer pool memory limit -
+* <u>Key</u>: `NSFS_BUF_POOL_MEM_LIMIT`  
+* <u>Type</u>: Number  
+* <u>Default</u>: 32MB = 33554432 bytes = 32 * 1024 * 1024  
+* <u>Description</u>: Set Buffer pool memory limit in order to increase performance. Suggested values 1-4GB, service restart required.  
```

Comment:
I think worth mentioning explicitly that we expect the number in bytes (also in "Buffer size").
```suggestion
* <u>Description</u>: Set Buffer pool memory limit (bytes) in order to increase performance. Suggested values 1-4GB, service restart required.  
```